js/src/components/google-ads-account-card/google-ads-account-card.js

```diff
@@ -7,32 +7,54 @@ import { __ } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
+import { GOOGLE_ADS_ACCOUNT_STATUS } from '.~/constants';
 import SpinnerCard from '.~/components/spinner-card';
 import useGoogleAccount from '.~/hooks/useGoogleAccount';
 import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
 import ConnectedGoogleAdsAccountCard from './connected-google-ads-account-card';
 import NonConnected from './non-connected';
 import AuthorizeAds from './authorize-ads';
-import { GOOGLE_ADS_ACCOUNT_STATUS } from '.~/constants';
+import DisabledCard from './disabled-card';
+import useGoogleAdsAccountStatus from '.~/hooks/useGoogleAdsAccountStatus';
 
 export default function GoogleAdsAccountCard() {
 	const {
 		google,
 		scope,
-		isResolving: isResolvingGoogleAccount,
+		hasFinishedResolution: hasResolvedGoogleAccount,
 	} = useGoogleAccount();
-	const { googleAdsAccount, isResolving: isResolvingGoogleAdsAccount } =
-		useGoogleAdsAccount();
 
-	if ( isResolvingGoogleAccount || isResolvingGoogleAdsAccount ) {
+	const {
+		googleAdsAccount,
+		hasFinishedResolution: hasResolvedGoogleAdsAccount,
+	} = useGoogleAdsAccount();
+	const {
+		hasAccess,
+		step,
+		hasFinishedResolution: hasResolvedAdsAccountStatus,
+	} = useGoogleAdsAccountStatus();
+
+	if (
+		! hasResolvedGoogleAccount ||
+		( ! hasResolvedGoogleAdsAccount && ! googleAdsAccount ) ||
+		! hasResolvedAdsAccountStatus
+	) {
 		return <SpinnerCard />;
 	}
 
+	if ( ! google || google.active === 'no' ) {
+		return <DisabledCard />;
+	}
+
 	if ( ! scope.adsRequired ) {
 		return <AuthorizeAds additionalScopeEmail={ google.email } />;
 	}
 
-	if ( googleAdsAccount?.status === GOOGLE_ADS_ACCOUNT_STATUS.DISCONNECTED ) {
+	if (
+		googleAdsAccount?.status === GOOGLE_ADS_ACCOUNT_STATUS.DISCONNECTED ||
```

Comment:
Thanks for catching that @eason9487 . Updated!

Reply:
Thanks for the update. Unfortunately, I'm not pretty sure if it actually fix the blank page issue and I think it's just reverted back to similar concerns in [my original comment in this thread](https://github.com/woocommerce/google-listings-and-ads/pull/2270#discussion_r1535143754):

-  > If the optional chaining operation `?.` here intends to handle the case of the `ads/connection` API failure, then this is still problematic for the overall `GoogleAdsAccountCard` component, because an uncaught error still occurs ~at `ConnectedGoogleAdsAccountCard` below~ within `NonConnected` component.
      <img width="510" alt="image" src="https://github.com/woocommerce/google-listings-and-ads/assets/17420811/815f10c1-fa51-4ded-874c-4d7731574679">
-  > However, when an error occurs in that API, it's more appropriate to limit it to rendering the `SpinnerCard` above instead of entering the flow below that `SpinnerCard`.

I have provided a way to simulate the issue in https://github.com/woocommerce/google-listings-and-ads/pull/2270#discussion_r1538908475. Could you double-confirm the next fix can:

1. Allow the page showing the error message to the user by the notification on the left-bottom of the page instead of a blank page,
2. When the `ads/connection` API fails, Limit it to showing the `SpinnerCard ` in this component instead of overflowing to any renders under the `SpinnerCard`.